hummingbot/market/kraken/kraken_order_book_tracker.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+import asyncio
+from collections import deque, defaultdict
+import logging
+import time
+from typing import (
+    Deque,
+    Dict,
+    List,
+    Optional
+)
+
+from hummingbot.logger import HummingbotLogger
+from hummingbot.core.data_type.order_book_tracker import (
+    OrderBookTracker,
+    OrderBookTrackerDataSourceType)
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.data_type.remote_api_order_book_data_source import RemoteAPIOrderBookDataSource
+from hummingbot.core.utils.async_utils import safe_ensure_future
+from hummingbot.market.kraken.kraken_api_order_book_data_source import KrakenAPIOrderBookDataSource
+from hummingbot.core.data_type.order_book_message import OrderBookMessage
+from hummingbot.core.data_type.order_book import OrderBook
+
+
+class KrakenOrderBookTracker(OrderBookTracker):
+    _krobt_logger: Optional[HummingbotLogger] = None
+
+    @classmethod
+    def logger(cls) -> HummingbotLogger:
+        if cls._krobt_logger is None:
+            cls._krobt_logger = logging.getLogger(__name__)
+        return cls._krobt_logger
+
+    def __init__(self,
+                 data_source_type: OrderBookTrackerDataSourceType = OrderBookTrackerDataSourceType.EXCHANGE_API,
+                 trading_pairs: Optional[List[str]] = None):
+        super().__init__(data_source_type=data_source_type)
+
+        self._order_book_diff_stream: asyncio.Queue = asyncio.Queue()
+        self._order_book_snapshot_stream: asyncio.Queue = asyncio.Queue()
+
+        self._ev_loop: asyncio.BaseEventLoop = asyncio.get_event_loop()
+        self._data_source: Optional[OrderBookTrackerDataSource] = None
+        self._saved_message_queues: Dict[str, Deque[OrderBookMessage]] = defaultdict(lambda: deque(maxlen=1000))
+        self._trading_pairs: Optional[List[str]] = trading_pairs
+
+    @property
+    def data_source(self) -> OrderBookTrackerDataSource:
+        if not self._data_source:
+            if self._data_source_type is OrderBookTrackerDataSourceType.REMOTE_API:
+                self._data_source = RemoteAPIOrderBookDataSource()
+            elif self._data_source_type is OrderBookTrackerDataSourceType.EXCHANGE_API:
+                self._data_source = KrakenAPIOrderBookDataSource(trading_pairs=self._trading_pairs)
+            else:
+                raise ValueError(f"data_source_type {self._data_source_type} is not supported.")
+        return self._data_source
+
+    @property
+    def exchange_name(self) -> str:
+        return "kraken"
+
+    async def start(self):
+        await super().start()
```

Comment:
All the stuff you do here are already done in the order book tracker base class's `start()` function. You should just remove the `start()` function from here altogether.